### PR TITLE
thrift to flatbuffers compiler and (de)serializer

### PR DESCRIPTION
**Pull request by paszea**
@jparise @gengmao

### CODE REVIEW

File: compiler/cpp/src/generate/t_flatb_generator.cc [hunk 1]
```diff
@@ -0,0 +1,449 @@
...
+string friendly_name(const string& type_name, int type_id, const string& strip_prefix) {
+  string fixed = type_name;
+  std::replace(fixed.begin(), fixed.end(), '.', '_');
+  if (fixed[0] == '[' && fixed[fixed.size() - 1] == ']') {
+      string stripped = fixed.substr(1, fixed.size() - 2);
+      if (stripped.substr(0, strip_prefix.size()) == strip_prefix) {
+        stripped = stripped.substr(strip_prefix.size());
+      }
+      string prefix;
+      if (type_id == 13) {
```
Comment (jparise) [inline compiler/cpp/src/generate/t_flatb_generator.cc]: Comment that 13 is `protocol::TType::T_MAP` (and `T_SET` below for 14).

File: lib/cpp/src/thrift/protocol/TFlatbProtocol.tcc [hunk 2]
```diff
@@ -0,0 +1,921 @@
...
+struct ReadContext {
+  ReadContext(const flatbuffers::Table* tbl, const reflection::Object* tbl_schema)
+      : table(tbl), table_schema(tbl_schema), table_tag(TABLE),
+        field_index(-1), elem_index(-1), vec_size(-1), vec_tag(LIST) { }
+
+  // The current table being read.
+  const flatbuffers::Table* table;
+
+  // The schema of the table.
+  const reflection::Object* table_schema;
```
Comment (jparise) [inline lib/cpp/src/thrift/protocol/TFlatbProtocol.tcc]: `table` and `table_schema` can be const pointers, too:

```cpp
const flatbuffers::Table * const table;
const reflection::Object * const table_schema;
```

File: lib/java/src/org/apache/thrift/protocol/FlatBufferSchemas.java [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+    }
+    return schema_.objects(field.type().index());
+  }
+
+  private String bfbs_filename_;
+  private ByteBuffer bfbs_file_content_;
+  private Schema schema_;
+
+  public static void main(String[] args) throws Exception {
+    FlatBufferSchemas schemas = new FlatBufferSchemas("/Users/xunliu/workplace/flatbuffers/t.bfbs");
```
Comment (jparise) [inline lib/java/src/org/apache/thrift/protocol/FlatBufferSchemas.java]: Maybe you don't want this hardcoded path in here?

**Comment by paszea** [comment]
addressed comments

**Pull request merged**